Mini Cart block - Fix the drawer content height to allow checkout button to show.

danielwrobert opened the pull request:
<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->

By giving the Mini Cart contents a `max-height` with the `-webkit-fill-available` property, we can ensure that the browser chrome (bottom address bar) is accounted for on mobile devices.

Props to @senadir and @Aljullu for their suggestions! 🎉 

<!-- Reference any related issues or PRs here -->

Fixes #7400 

<!-- Don't forget to update the title with something descriptive. -->
<!-- If you can, add the appropriate labels -->

### Screenshots

<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->

| Before | After |
| ------ | ----- |
|   ![08F40EA6-FB9A-49A7-B8AF-0C972500F3EC](https://user-images.githubusercontent.com/481776/215877738-4299ed05-12de-4397-a149-a819810d9d5c.PNG)    |    ![IMG_2580](https://user-images.githubusercontent.com/481776/215877742-13af250c-543e-4c7f-b25c-f01b03eefae9.PNG)   |
|    ![569A73FD-FB86-43B5-AE22-3B00C416A037](https://user-images.githubusercontent.com/481776/215877740-121e3da0-fefb-4536-ab1e-ff1253cf77dd.PNG)    |   ![IMG_2581](https://user-images.githubusercontent.com/481776/215877744-973a6670-3fd5-46e6-b3f0-43cd221a3860.PNG)    |

### Testing
<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1. Check out branch and run `npm start`.
2. Set up the Mini Cart in your global navigation, if you don't have it already.
3. View the front end of your site from a mobile device (or Xcode Simulator).
4. Add a product to your cart.
5. Open the Mini Cart drawer.
6. Confirm the checkout button is fully visible (see After screenshots above).

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Performance Impact

<!-- Please document any known performance impact (positive or negative) here. If negative, provide some rationale for why this is an okay tradeoff or how this will be addressed. -->

### Changelog

> Mini Cart block: Fix the drawer content height to allow the checkout button to be visible.

danielwrobert committed:
Add max-height to Mini Cart drawer.

By giving the Mini Cart contents a max height with the
-webkit-fill-available property, we can ensure that the browser chrome
is accounted for on mobile devices.

Aljullu reviewed (commented):
I was testing this PR in Firefox and Brave for Android and it doesn't seem to work for me, the URL is still showing above the button.

However, I could fix it changing `height: 100vh` to [dynamic units](https://developer.mozilla.org/en-US/docs/Web/CSS/length#dynamic): `height: 100dvh`.

danielwrobert committed:
Add `dvh` and keep `vh` as a fallback for height.

By adding `dvh`, we also account for non-webkit mobile browsers that
have the same hidden button issue.

See
https://github.com/woocommerce/woocommerce-blocks/pull/8351#pullrequestreview-1278867767

Keeping `vh` as a fallback, along with `-webkit-fill-available` gives us
the widest range of support.

danielwrobert commented:
> I was testing this PR in Firefox and Brave for Android and it doesn't seem to work for me, the URL is still showing above the button.
> 
> However, I could fix it changing `height: 100vh` to [dynamic units](https://developer.mozilla.org/en-US/docs/Web/CSS/length#dynamic): `height: 100dvh`.

@Aljullu ah right, because those aren't webkit-based browsers and I was only targeting webkit here. Thanks for catching and for the suggestion to use `dvh`!

Instead of replacing, I've added a `height: 100dvh` after the `height: 100vh` so that browsers that don't yet support `dvh` will still use `vh` as a fallback. This seems to give us the widest range of support.

Aljullu reviewed (approved):
Tested again and it's working well on my end. :ship:
